### PR TITLE
changed voting to use forms

### PR DESCRIPTION
**Pull request by codiebeulaine**

### CODE REVIEW

**Comment by codiebeulaine** [comment]
@miltontony ready for review

File: polls/forms.py [hunk 1]
```diff
@@ -1,5 +1,18 @@
...
+
+class VoteForm(forms.Form):
+    choice = forms.MultipleChoiceField(required=True)
+
+    def __init__(self, *args, **kwargs):
+        super(VoteForm, self).__init__(*args, **kwargs)
+        self.fields['choice'].choices = [(
+            c.pk, c.title) for c in Choice.objects.all()]
+
+    def clean_choice(self):
```
Comment (miltontony) [inline polls/forms.py]: You don't need this anymore.

File: polls/views.py [hunk 2]
```diff
@@ -49,36 +48,38 @@ def poll_results(request, poll_id):
...
+
+    def get_context_data(self, *args, **kwargs):
+        context = super(
+            VoteView, self).get_context_data(*args, **kwargs)
+        question_id = self.kwargs.get('question_id')
+        question = get_object_or_404(Question, pk=question_id)
+        context.update({'question': question})
+        return context
+
+    def form_invalid(self, form, *args, **kwargs):
```
Comment (miltontony) [inline polls/views.py]: Not needed anymore

File: tuneme/templates/polls/detail.html [hunk 3]
```diff
@@ -9,7 +9,7 @@
 	<div class="icon"><img src="{% static 'img/polls.png' %}" alt="Have Your Say"></div>
 	<h6><a href="">{% trans "Have Your Say" %}</a></h6>
 </div>
-
+{{form.errors}}
```
Comment (miltontony) [inline tuneme/templates/polls/detail.html]: Not needed. You'll need to change the error below on `line 15` to display the error using `form.choice.errors`

File: polls/tests/test_voting.py [hunk 4]
```diff
@@ -75,7 +75,7 @@ def test_voting_once_only(self):
         self.assertContains(response, 'is this a test')
         response = client.post(reverse('molo.polls:vote',
                                kwargs={'question_id': question.id}))
-        self.assertContains(response, "You didn&#39;t select a choice")
+        self.assertContains(response, "field is required")
```
Comment (justinvdm) [inline polls/tests/test_voting.py]: A bit bikesheddy, but `You didn't select a choice` seems a bit more descriptive an error than `... field is required` for users to see. Should we maybe keep that error message the same?

File: polls/views.py [hunk 5]
```diff
@@ -2,13 +2,12 @@
 from django.http import HttpResponseRedirect
 from django.core.urlresolvers import reverse
 from django.views import generic
-from django.utils.translation import ugettext_lazy as _
+# from django.utils.translation import ugettext_lazy as _
```
Comment (justinvdm) [inline polls/views.py]: Should be maybe just remove this line? We'll still have it around somewhere in our commit history.

File: polls/tests/test_voting.py (hunk 4 above)
Comment (miltontony) [inline polls/tests/test_voting.py]: I had the same feeling about this. @codiebeulaine you can specify a custom error message for the required message in forms.py using [Django's error_messages option](https://docs.djangoproject.com/es/1.9/ref/forms/fields/#error-messages)

File: polls/views.py (hunk 5 above)
Comment (miltontony) [inline polls/views.py]: +1 on removing

File: polls/tests/test_voting.py (hunk 4 above)
Comment (codiebeulaine) [inline polls/tests/test_voting.py]: @miltontony should I keep you didn't select a choice or display the error using form.choice.errors?

File: polls/views.py [hunk 6]
```diff
@@ -49,36 +48,38 @@ def poll_results(request, poll_id):
...
                 choice.votes = F('votes') + 1
                 choice.choice_votes.add(obj)
                 choice.save()
-            return HttpResponseRedirect(reverse('molo.polls:results',
-                                                args=(question.id,)))
-        else:
-            return render(request, 'polls/detail.html', {
-                'question': question,
-                'error_message': _("You are only allowed to vote once."),
-            })
```
Comment (justinvdm) [inline polls/views.py:81]: I would imagine we'd still want to disallow re-voting, just checking if we are intentionally removing this bit of logic here?

Comment (justinvdm) [inline polls/views.py:81]: s/disallow re-voting/display some message to the user if they attempt to re-vote

File: polls/tests/test_voting.py (hunk 4 above)
Comment (miltontony) [inline polls/tests/test_voting.py]: @codiebeulaine both. Keep the "you didn't select" error message and use `form.choice.errors` to show the error message

File: polls/views.py (hunk 6 above)
Comment (miltontony) [inline polls/views.py:81]: Intentionally removing. Essentially a user can only vote once, but we just won't be showing them the message.

**Comment by justinvdm** [comment]
Left a few comments, looks good.

**Comment by codiebeulaine** [comment]
@miltontony @justinvdm ready for review

File: polls/forms.py [hunk 7]
```diff
@@ -1,5 +1,17 @@
...
 
 
 class TextVoteForm(forms.Form):
     answer = forms.CharField(required=True)
+
+
+class VoteForm(forms.Form):
+    choice = forms.MultipleChoiceField(
+        required=True,
+        error_messages={'required': "You didn't select a choice"})
```
Comment (justinvdm) [inline polls/forms.py]: Do we need to wrap the string in a `_` for translations?

**Comment by justinvdm** [comment]
:+1: from me

**Pull request merged**